app/components/edit_resource_button.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+# EditResourceButton
+# edit collection btn + update work btn
+# pass in resource instead of work_version and infer type (collection or work_version)
+
+class EditResourceButton < ApplicationComponent
+  attr_reader :resource, :policy
+
+  def initialize(resource:, policy:)
+    @resource = resource
+    @policy = policy
+  end
+
+  private
+
+    def path
+      if collection?
+        dashboard_form_collection_details_path(resource.id)
+      elsif has_draft?
+        # update the existing draft version
+        dashboard_form_work_version_details_path(resource.work.draft_version.id)
+      elsif policy.new?
+        # create a new draft work version
+        dashboard_work_work_versions_path(resource.work)
+      end
+    end
+
+    def label
+      collection? ? I18n.t('resources.collection.edit_button') : I18n.t('resources.work_version.edit_button')
+    end
+
+    def method
+      return if collection? || has_draft?
+
+      'post'
+    end
+
+    def has_draft?
+      resource.work.draft_version
+    end
+
+    def type
+      resource.class.name
+    end
+
+    def collection?
+      type == 'CollectionDecorator'
```

Comment:
👍 - FWIW, this code was copied from `DeleteResourceButtonComponent`. I agree that this approach is cleaner and simpler.

Reply:
Just goes to show, you can't trust @rschenk. (joke) Actually, he used `type` elsewhere in the class, so that makes sense. I don't think we're doing that here? So we can take a short-cut.

Reply:
![hacking in progress](https://media.giphy.com/media/10FwycrnAkpshW/giphy.gif)